core/unit_test/TestPolicyConstruction.hpp

```diff
@@ -846,6 +1154,103 @@ void more_md_range_policy_construction_test() {
                                                                {T(2), T(2)}};
 }
 
+TEST(TEST_CATEGORY, md_range_policy_compile_time_deduction_guides) {
+  Kokkos::DefaultExecutionSpace des{};
+  ImplicitlyConvertibleToDefaultExecutionSpace notEs{};
+  SomeExecutionSpace ses{};
+  TEST_EXECSPACE es{};
+
+  constexpr int t[5]       = {};
+  constexpr int64_t tt[5]  = {};
+  Kokkos::Array<int, 3> a  = {};
+  Kokkos::Array<int, 2> aa = {};
```

Comment:
```suggestion
  Kokkos::Array<int64_t, 3> a  = {};
  Kokkos::Array<int64_t, 2> aa = {};
```

Reply:
Reverted this change back as it wasn't the issue